### PR TITLE
updated auth components with props to control all available redirects

### PR DESCRIPTION
**Pull request by 00salmon**
## Description of the change

Update Auth components to allow a dual portal configuration. Increases the customizability of routing 

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

 - Jira issue number: https://saasquatch.atlassian.net/browse/DEV-3171
 - Process.st launch checklist: https://app.process.st/runs/DEV-3171-Component-Boilerplate-and-mint-update-o3odNUy7pswnNgLZQ4RPtA

### Development

- [x] [Prettier](https://www.npmjs.com/package/prettier) was run
- [x] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [x] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has a Jira number
- [ ] This pull request has a Process.st launch checklist
- [ ] "Ready for review" label attached to the PR and reviewers pinged on Slack

### Code review 

- [ ] Changes have been reviewed by at least one other engineer
- [ ] Security impacts of this change have been considered

### CODE REVIEW

**Comment by DerekSiemens** [comment]
DS: Flagging need for changelog and spec updates

File: packages/mint-components/src/components/sqm-portal-login/usePortalLogin.tsx [hunk 1]
```diff
@@ -46,6 +46,8 @@ export function usePortalLogin({ nextPage }) {
     states: {
       loading,
       error: errorMessage,
+      registerPath:props.registerPath,
+      forgotPasswordPath:props.registerPath
```
Comment (DerekSiemens) [inline packages/mint-components/src/components/sqm-portal-login/usePortalLogin.tsx:50]: Flagging bug, need to return `props.forgotPasswordPath`

Comment (00salmon) [inline packages/mint-components/src/components/sqm-portal-login/usePortalLogin.tsx:50]:
```suggestion
      forgotPasswordPath:props.forgotPasswordPath
```

**Comment by DerekSiemens** [comment]
I have updated specs and pulled some historical ones in where existing specs were missing

File: packages/mint-components/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@saasquatch/mint-components",
-  "version": "1.3.0",
+  "version": "1.3.3-7",
```
Comment (DerekSiemens) [inline packages/mint-components/package.json:3]: I think this would actually become 1.3.2 as this PR will likely go out before loyalty components

Comment (loganvolkers) [reply packages/mint-components/package.json:3]: This should actually be a `1.4.0` since it introduces new functionality.

Comment (DerekSiemens) [reply packages/mint-components/package.json:3]: Edit: Lets bump to 1.4.0

File: packages/mint-components/src/components/sqm-big-stat/BigStat.feature [hunk 3]
```diff
@@ -157,3 +157,24 @@ Feature: Big Stat
       | /referralsCount           | { programId_eq: programId }                              |
       | /referralsCount/converted | { programId_eq: programId, dateConverted_exists: true }  |
       | /referralsCount/started   | { programId_eq: programId, dateConverted_exists: false } |
+
+  Scenario: By default program context is used to source the programId used for queries
```
Comment (loganvolkers) [inline packages/mint-components/src/components/sqm-big-stat/BigStat.feature:161]:
```suggestion
  Scenario: Program context is used by default to source the programId used for queries
```

File: packages/mint-components/package.json [hunk 4]
```diff
@@ -56,7 +56,7 @@
     "workbox-build": "4.3.1"
   },
   "dependencies": {
-    "@saasquatch/component-boilerplate": "^1.2.2",
+    "@saasquatch/component-boilerplate": "^1.2.3-11",
```
Comment (loganvolkers) [inline packages/mint-components/package.json:59]: Should use latest version?

```suggestion
    "@saasquatch/component-boilerplate": "^1.3.0",
```

**Review by loganvolkers — DISMISSED** [review]
Looks good to me

**Review by DerekSiemens — APPROVED** [review]

**Pull request merged**